src/xtgeo/well/well1.py

```diff
@@ -1132,10 +1130,7 @@ def may_overlap(self, other):
 
         if xmin1 > xmax2 or ymin1 > ymax2:
             return False
-        if xmin2 > xmax1 or ymin2 > ymax1:
-            return False
-
-        return True
+        return not (xmin2 > xmax1 or ymin2 > ymax1)
```

Comment:
Why not the first if? (at line 1131)

And same here - not straightforward to "translate in head" (line 1133)

Reply:
I am not sure. Not really a fan of this ruff rule either.